### PR TITLE
Replace addressing.CiliumIPv{4,6} by netip.Addr type

### PR DESCRIPTION
**Pull request by tklauser**
Review by commit. The main change is in the 2nd to last commit switching the `endpoint.Endpoint` type to use `netip.Addr`. Its commit message is as follows:

    endpoint: convert Endpoint type from addressing.CiliumIPv{4,6} to netip.Addr
    
    A netip.Addr always uses 24 bytes, regardless of the address type. An
    addressing.CiliumIPv4 uses 28 bytes (24 bytes for the slice header and 4
    bytes for the address). An addressing.CiliumIPv6 uses 56 bytes (24 bytes
    for the slice header and 32 bytes for the address). Also see [1] for
    more details on further advantages of the netip types in terms of
    performance.
    
    [1] https://tailscale.com/blog/netaddr-new-ip-type-for-go/
    
    This change thus slightly reduces the in-memory size of the Endpoint
    type and also allows to drop the custom addressing types in favor of Go
    standard library types with corresponding helper functions and methods.

### CODE REVIEW

**Review by gandro — APPROVED** [review]
Very nice work!

File: pkg/datapath/linux/config/config_test.go [hunk 1]
```diff
@@ -38,8 +39,8 @@ var (
 	dummyNodeCfg  = datapath.LocalNodeConfiguration{}
 	dummyDevCfg   = testutils.NewTestEndpoint()
 	dummyEPCfg    = testutils.NewTestEndpoint()
-	ipv4DummyAddr = []byte{192, 0, 2, 3}
-	ipv6DummyAddr = []byte{0x20, 0x01, 0xdb, 0x8, 0x0b, 0xad, 0xca, 0xfe, 0x60, 0x0d, 0xbe, 0xe2, 0x0b, 0xad, 0xca, 0xfe}
+	ipv4DummyAddr = netip.MustParseAddr("192.0.2.3")
```
Comment (aditighag) [inline pkg/datapath/linux/config/config_test.go:42]: I was wondering why this wouldn't return an error. Nice utility function for tests usage!

**Review by aditighag — APPROVED** [review]
I only looked at "endpoint:" commits codeowner. Nice improvement! 🚀

**Comment by joestringer** [comment]
^ Since @aditighag only reviewed for @ciluum/endpoint, I re-requested review from the other codeowners.

**Review by joestringer — APPROVED** [review]
:tada: LGTM for my codeowners

**Review by christarazi — APPROVED** [review]
Very satisfying PR, thanks!

**Comment by tklauser** [comment]
Reviews for all codeowners are in and tests passed. Marking ready-to-merge.

**Pull request merged**